cpg_infra/config.py

```diff
@@ -0,0 +1,261 @@
+# pylint: disable=missing-function-docstring,missing-class-docstring,invalid-name
+"""
+This module contains all the configuration objects that are used to
+describe the CPG infrastructure, including what's required from a
+specific dataset.
+"""
+
+import dataclasses
+from enum import Enum
+from types import UnionType
+import toml
+
+
+class DeserializableDataclass:
```

Comment:
Is serialization necessary for storing the Pulumi state? Would be good to explain here why this rather complicated logic is necessary.